### PR TITLE
Merge 2.10.169.0 to 3.0

### PR DESCRIPTION
**Pull request by MMcM**
Conflicts (resolved by hand):
* `docs/ReleaseNotes.md`
* `fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordCursor.java`
* `fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryIntersectionPlan.java`

### CODE REVIEW

**Comment by MMcM** [comment]
test this please

**Comment by alecgrieser** [comment]
Did you need to resolve any merge conflicts, or did this merge cleanly?

**Comment by alecgrieser** [comment]
Oh, I see--you left a comment. Sorry

File: docs/ReleaseNotes.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Another, smaller change that has been made is that by default, new indexes added
 // begin next release
 ### NEXT_RELEASE
 
-* **Bug fix** Fix LiteralKeyExpression hash code [(Issue #1066)](https://github.com/FoundationDB/fdb-record-layer/issues/1066)
+* **Bug fix** Fix 1 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
```
Comment (alecgrieser) [inline docs/ReleaseNotes.md:19]: I guess there's a slight concern that if there's not a 2.10 release between now and when 3.0 is released, this bug fix will never get a release note. I think we'll probably want to do one last 2.10 release (if there's anything not yet released) before cutting 3.0, though, so probably not a real concern.

**Review by alecgrieser — APPROVED** [review]

**Pull request merged**